packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswer/e2e/pageObject.ts

```diff
@@ -0,0 +1,295 @@
+import {
+  isUaClickEvent,
+  isUaCustomEvent,
+} from '../../../../../../playwright/utils/requests';
+
+const minimumCitationTooltipDisplayDurationMs = 1500;
+
+export class GeneratedAnswerObject {
+  constructor(
+    public page: Page,
+    public streamId: string
+  ) {
+    this.page = page;
+    this.streamId = streamId;
+  }
+
+  get likeButton(): Locator {
+    return this.page.locator('[data-cy="feedback__like-button"]');
+  }
+
+  get dislikeButton(): Locator {
+    return this.page.locator('[data-cy="feedback__dislike-button"]');
+  }
+
+  get copyToClipboardButton(): Locator {
+    return this.page.locator(
+      '[data-test="generated-answer__copy-to-clipboard"]'
+    );
+  }
+
+  get toggleButton(): Locator {
+    return this.page.locator('[data-test="generated-answer__toggle-button"]');
+  }
+
+  feedbackOption(value: string, index: number): Locator {
+    return this.page
+      .locator('.feedback-modal-qna__question input', {
+        has: this.page.locator(`[value="${value}"]`),
+      })
+      .nth(index);
+  }
+
+  questionContainer(questionText: string): Locator {
+    return this.page.locator('.feedback-modal-qna__question', {
+      hasText: questionText,
+    });
+  }
+
+  answerOption(questionText: string, answerValue: string): Locator {
+    return this.questionContainer(questionText).locator('.slds-radio_button', {
+      hasText: new RegExp(`^${answerValue}$`),
+    });
+  }
+
+  get feedbackDocumentUrlInput(): Locator {
+    return this.page.locator(
+      '.feedback-modal-qna__body input[name="documentUrl"]'
+    );
+  }
+
+  get feedbackDetailsInput(): Locator {
+    return this.page.locator(
+      '.feedback-modal-qna__body [data-name="details"] textarea'
+    );
+  }
+
+  get submitFeedbackButton(): Locator {
+    return this.page.getByRole('button', {name: /Send feedback/i});
+  }
+
+  get citationLink(): Locator {
+    return this.page.locator(
+      '[data-test="generated-answer__citations"] .citation__link'
+    );
+  }
+
+  async hoverOverCitation(index: number): Promise<void> {
+    // waiting 500ms to allow the component to render completely, cause any re-rendering abort the hover action.
+    await this.page.waitForTimeout(500);
+    await this.citationLink.waitFor({state: 'visible'});
+    await this.citationLink.nth(index).hover();
+    await this.page.waitForTimeout(minimumCitationTooltipDisplayDurationMs);
+    await this.page.mouse.move(0, 0);
+  }
+
+  async clickOnCitation(index: number): Promise<void> {
+    await this.citationLink.nth(index).click();
+  }
+
+  async typeInFeedbackDocumentUrlInput(text: string): Promise<void> {
+    await this.feedbackDocumentUrlInput.fill(text);
+  }
+
+  async typeInFeedbackDetailsInput(text: string): Promise<void> {
+    await this.feedbackDetailsInput.fill(text);
+  }
+
+  async clickSubmitFeedbackButton(): Promise<void> {
+    await this.submitFeedbackButton.click();
+  }
+
+  async fillFeedbackForm(answers: Record<string, string>): Promise<void> {
```

Comment:
This is kinda heavily relying on the fact that the text of the questions won't change right?
Wouldn't it be better to either use their "tags"? (I think we have the tag "relevant" "hallucination" etc. on each questions.)
Or maybe simply accept just a list of answers and fill them in order?

Reply:
Addressed 👌